lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -135,3 +152,36 @@ hashSeed = serialize . blake2b256 . serialize
 
 blake2b256 :: ScrubbedBytes -> ScrubbedBytes
 blake2b256 = BA.convert . hash @ScrubbedBytes @Blake2b_256
+
+-- | Compute passphrase as hash of the root public key.
+deriveHDPassphrase :: Key 'RootK XPub -> Passphrase "random"
+deriveHDPassphrase (Key pk) = Passphrase $
+    PBKDF2.generate
+        (PBKDF2.prfHMAC SHA512)
+        (PBKDF2.Parameters
+             500 -- Parameters for the hashing function. 500 iter of PBDKF2 with HMAC-SHA256
+             passLen)
+        (unXPub pk)
+        ("address-hashing"::ByteString)
+  where
+    -- Password length in bytes
+    passLen = 32
+
+eskToHdPassphrase :: Key 'RootK XPrv -> Passphrase "random"
+eskToHdPassphrase = deriveHDPassphrase . publicKey
```

Comment:
Please no :sweat_smile:  ... This `esk` comes from the legacy terminology and is quite confusing in the new codebase where there's no clear mention of `esk` anywhere.
Plus, I'd rather keep the module API "orthogonal" and smaller, hence simply have a:

```hs
hdPassphrase :: Key 'Root XPub -> Passphrase "derivation-path" 
```

I don't think this is a common enough operation to deserve an ad-hoc helper that does the composition. The composition can simply occur at the call-site. Don't you think ?

Reply:
I resign with introduction of `eskToHdPassphrase`, just used `publicKey` to realize `Key 'Root XPriv -> Key 'Root XPub`